components/style/parallel.rs

```diff
@@ -153,7 +169,8 @@ fn top_down_dom<'a, 'scope, E, D>(nodes: &'a [SendNode<E::ConcreteNode>],
     {
         // Scope the borrow of the TLS so that the borrow is dropped before
         // a potential recursive call when we pass TailCall.
-        let mut tlc = tls.ensure(|| traversal.create_thread_local_context());
+        let mut tlc = tls.ensure(
+            |slot: &mut RefMut<Option<D::ThreadLocalContext>>| create_thread_local_context(traversal, slot));
```

Comment:
Though I think you may be able to just write |slot| and have type inference do its thing.